integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java

```diff
@@ -84,47 +92,66 @@
    * @param configuration
    * @throws IOException
    */
-  private static void populateCarbonTable(Configuration configuration, String paths)
+  private CarbonTable populateCarbonTable(Configuration configuration)
       throws IOException {
-    String dirs = configuration.get(INPUT_DIR, "");
-    String[] inputPaths = StringUtils.split(dirs);
-    String validInputPath = null;
+    TableInfo tableInfo = getTableInfo(configuration);
+    CarbonTable carbonTable = null;
+    if (tableInfo != null) {
+      carbonTable = CarbonTable.buildFromTableInfo(tableInfo);
+      CarbonMetadata.getInstance().addCarbonTable(carbonTable);
+      return carbonTable;
+    }
+    String inputDir = configuration.get(INPUT_DIR, "");
+    String[] inputPaths = StringUtils.split(inputDir);
     if (inputPaths.length == 0) {
       throw new InvalidPathException("No input paths specified in job");
-    } else {
-      if (paths != null) {
-        for (String inputPath : inputPaths) {
-          if (paths.startsWith(inputPath)) {
-            validInputPath = inputPath;
-            break;
-          }
-        }
-      }
     }
+    Arrays.sort(inputPaths);
```

Comment:
because there are serveral paths, the sortest path are the table path, so I sort all and then get the first one